cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go

```diff
@@ -68,6 +68,13 @@ func (s *Server) listReleasesInCluster(ctx context.Context, namespace string) ([
 		return nil, err
 	}
 
+	// TODO (gfichtenholt)
+	// 1) we need to make sure that .List() always returns results in the
+	// same order, otherwise pagination is broken (duplicates maybe returned and some entries
+	// missing).
```

Comment:
I'm still not clear on this: isn't your point (1) fixed by point (2)? That is, AIUI, if we update to pass the token used for pagination from the k8s api server, then we shouldn't be getting any duplicates or missing entries even when there is churn on the data level (ie. new HelmReleases or deleted ones).

Ah, or is the issue that the plugin is currently fetching all the objects and then returning just a subset for the paginated results of our server (in which case, we won't have the snapshot consistency, since each request to the kubeapps API will be getting results from a different snapshot?